include/eve/module/algo/algo/transform_copy_if.hpp

```diff
@@ -0,0 +1,168 @@
+          auto [vals, mask] = func(loaded);
+          using out_t = eve::element_type_t<decltype(vals)>;
+          auto cvt_and_store_it = views::convert(of, eve::as<out_t>{});
+          auto write_end = eve::compress_store[eve::unsafe](vals, mask, cvt_and_store_it);
+          of += write_end - cvt_and_store_it;
+          return false;
+        }
+
+        EVE_FORCEINLINE
+        bool step_2(auto f)
+        {
+          // ol - of < cardinal
+          auto out_ignore   = eve::keep_first(ol - of);
+          auto loaded       = eve::load(f);
+          auto [vals, mask] = func(loaded);
+          using out_t = eve::element_type_t<decltype(vals)>;
+          auto cvt_and_store_it = views::convert(of, eve::as<out_t>{});
+          auto write_end = eve::compress_store[eve::unsafe][eve::ignore_none][out_ignore](vals, mask, cvt_and_store_it);
+          of += write_end - cvt_and_store_it;
+          return of == ol;
+        }
+      };
+
+      template<relaxed_range In, relaxed_range Out, typename Func>
+      EVE_FORCEINLINE
+      auto operator()(In&& in, Out&& out, Func func) const -> unaligned_iterator_t<Out>
+      {
+        if( in.begin() == in.end() || out.begin() == out.end() )
+          return out.begin();
+
+        auto [processed_in, processed_out] = temporary_preprocess_ranges_hack(
+          TraitsSupport::get_traits(), in, out);
+
+        auto iteration = two_stage_iteration(processed_in.traits(), processed_in.begin(), processed_in.end());
+
+        auto of = unalign(processed_out.begin());
+        auto ol = unalign(processed_out.end());
+
+        delegate<decltype(of), Func> d {of, ol, func};
+        iteration(d);
+        return eve::unalign(out.begin()) + (d.of - processed_out.begin());
+      }
+
+      template<relaxed_range In, relaxed_range Out, typename Op, typename P>
+      EVE_FORCEINLINE
+      auto operator()(In&& in, Out&& out, Op op, P p) const -> unaligned_iterator_t<Out>
+      {
+        return operator()(in, out, [&](auto elt) {
+          return kumi::make_tuple(op(elt), p(elt));
+        });
+      }
+    };
+  }
+
+  //==============================================================================================
+  //! @addtogroup algos
+  //! @{
+  //!   @var transform_copy_if
+  //!
+  //!   **Defined in header**
+  //!
+  //!   @code
+  //!   #include <eve/module/algo.hpp>
+  //!   @endcode
+  //!
+  //!   @brief Similar to applying `eve::transform_to` and `eve::copy_if` at the same time.
+  //!
+  //!   Conditionally copies values from an input range to an output range,
+  //!   transforming them in the process.
+  //!
+  //!   If the output range is too small, fills all the available space and then stops.
+  //!   
+  //!   If the output range's element type is different from the type of the values returned by the transforming function, performs the appropriate conversions.
+  //!
+  //!   There are two overloads: one that takes in two separate operation and predicate functions,
```

Comment:
I don't think we want this one. Is it really beneficial?

Reply:
the version with two functions still seemed to be faster (?) and it was convenient when testing things out, but if you think it's not a good interface choice we can remove it

Reply:
> still seemed to be faster

It's a bug somewhere if it is, Shouldn't be. (Probably measurement issue).
I don't think we want it.